### DISCUSSION TITLE
fallback template for null/blank values

### DISCUSSION THREAD

**Discussion by paulRbr** [microcks/microcks#1424] [Q&A] [answered]
Hi there,

I have a question regarding the [mock templates](https://microcks.io/documentation/references/templates/). Is there a way to add a fallback mock if the context expression returns a null / blank value?

e.g. for mocking a POST response we use the `request.body` values in [this example](https://github.com/bump-sh-examples/train-travel-api/blob/be2c394a7d8be29eaf5addb118206d2d46400731/.microcks/overlays.yaml#L141-L142) but as those parameters are not required in the request they might be empty. Thus turning the mock response in the following invalid JSON format:
```json
{
  "id": "af64322f-8726-4252-af37-c1bd5d3db41d",
  "trip_id": "4d67459c-af07-40bb-bb12-178dbb88e09f",
  "passenger_name": "john",
  "has_bicycle": ,
  "has_dog": ,
  "links": {
    "self": "https://api.example.com/bookings/af64322f-8726-4252-af37-c1bd5d3db41d"
  }
}
```
Notice the empty values in front of `has_bicycle` and `has_dog`?

My question is thus: is there a way to write a “fallback” expression something like `{{ request.body/has_bicycle  | randomBoolean() }}`?

**Comment by lbroudoux** [comment, marked as answer]
Hi @paulRbr 

Thanks for raising the discussion - that's a fair point and we don't have any fallback for null/blank values as of today. That would be an interesting enhancement! Would you mind creating an issue in https://github.com/microcks/microcks/issues to track this one?

**Reply by lbroudoux** [reply]
Any free cycle to create this issue? 😉

**Reply by paulRbr** [reply]
Sorry was on my phone when I say your answer, and got sidetracked. Here it is: https://github.com/microcks/microcks/issues/1446
